charm-slurmctld/src/charm.py

```diff
@@ -495,6 +497,16 @@ def _on_write_slurm_config(self, event):
             event.defer()
             return
 
+        # check if both certificates are supplied
+        tls_key = self.model.config['tls-key']
+        tls_cert = self.model.config['tls-cert']
+        self._stored.use_tls = (bool(tls_key) and bool(tls_cert))
+        self._stored.use_tls_ca = bool(self.model.config['tls-ca-cert'])
+
+        # TODO this will fire everytime a the configuration changed, we don't
+        #      need that. This should happen only if the tls configs changed
+        self._etcd.setup_tls()
```

Comment:
to implement the TODO, thw following could be done:

```suggestion
        # check if both certificates are supplied and setup TLS in case they are
        tls_key = self.model.config['tls-key']
        tls_cert = self.model.config['tls-cert']
        if not self._stored.use_tld and not self._stored.use_tls_ca:
            self._stored.use_tls = (bool(tls_key) and bool(tls_cert))
            self._stored.use_tls_ca = bool(self.model.config['tls-ca-cert'])
            if self._stored.use_tld and self._stored.use_tls_ca:
                self._etcd.setup_tls()
```

A bit strange? yes, but I guess the logic is it

Reply:
We have 3 different states:
- no TLS
- only TLS certs
- TLS certs + CA cert

We only need to run the function `setup_tls` when changing from one state to another, we have 6 different transitions, so we need a logic a bit more complex than this one.

Reply:
This is not critical at all, the `setup_tls` function has a safeguard. This task took longer than I anticipated, I don't think it is worth implementing this logic right now.

Reply:
This is not a problem at all, `setup_tls()` will not cause any harm. I don't think we should worry about this right now.